stdlib/public/Differentiation/FloatingPointDifferentiation.swift.gyb

```diff
@@ -282,4 +282,28 @@ where
     let y = squareRoot()
     return (y, { v in v / (2 * y) })
   }
+
+  @inlinable
+  @derivative(of: minimum)
+  static func _vjpMinimum(_ x: Self, _ y: Self) -> (
+    value: Self, pullback: (TangentVector) -> (TangentVector, TangentVector)
+  ) {
+    func pullback(_ v: TangentVector) -> (TangentVector, TangentVector) {
```

Comment:
This pullback is defined according to the logic of [`FloatingPoint.minimum`](https://github.com/apple/swift/blob/main/stdlib/public/core/FloatingPoint.swift#L1700-L1704) so that the corner cases for returning either `x` or `y` are respected.

Reply:
This implementation is running the identical control flow in the VJP and the pullback. It should be simplified so that the pullback is just straight line code and the control flow is executed only once. You can do:

```swift
@inlinable
@derivative(of: minimum)
static func _vjpMinimum(_ x: Self, _ y: Self) -> (
  value: Self, pullback: (TangentVector) -> (TangentVector, TangentVector)
) {
  if x <= y || y.isNaN { return (x, { v in (v, .zero) }) }
  return (y, { v in (.zero, v) })
}
```

The same suggestion applies to `maximum`.

Reply:
Ah I see.. yes, thanks for the remark @rxwei .. i am fixing that.

Reply:
fixed in bc40409.

Reply:
Thanks.